### PR TITLE
[Core] Shorten the membership checking time to 5 seconds.

### PR DESCRIPTION
**Pull request by rkooo567**
## Why are these changes needed?

60 seconds are too long in case the GCS is restarted. This can unblock the problem when the GCS is restarted, raylets won't be killed. We will discuss the fundamental fix and follow up with it. 

## Related issue number

Address https://github.com/ray-project/ray/issues/34763 short term

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by fishbone — APPROVED** [review]

File: src/ray/common/ray_config_def.h [hunk 1]
```diff
@@ -267,7 +267,7 @@ RAY_CONFIG(int64_t, fetch_warn_timeout_milliseconds, 60000)
 /// object directory.
 RAY_CONFIG(int64_t, fetch_fail_timeout_milliseconds, 600000)
 
-/// Temporary workaround for https://github.com/ray-project/ray/pull/16402.
+/// Temporary workaround for https://gitsb.com/ray-project/ray/pull/16402.
```
Comment (iycheng) [inline src/ray/common/ray_config_def.h:270]: Could you revert this back?

Comment (rkooo567) [reply src/ray/common/ray_config_def.h:270]: didn't notice that lol.. will do

**Comment by jjyao** [comment]
What's the issue? It's not clear to me from the PR description.

**Comment by rkooo567** [comment]
@jjyao https://github.com/ray-project/ray/issues/34763 <- this is the context!

**Comment by rkooo567** [comment]
But it is not a fundamental fix, so it won't close the issue.

**Comment by jjyao** [comment]
Left some comments on the issue. Still trying to understand the problem.

**Comment by rkooo567** [comment]
I am merging this PR as a temporary workaround.

**Pull request merged**